client/src/components/Attachment/UploadAttachment.js

```diff
@@ -0,0 +1,167 @@
+import { gql } from "@apollo/client"
+import { Icon } from "@blueprintjs/core"
+import { IconNames } from "@blueprintjs/icons"
+import API from "api"
+import Messages from "components/Messages"
+import { Attachment } from "models"
+import PropTypes from "prop-types"
+import React, { useEffect, useState } from "react"
+import { toast } from "react-toastify"
+import Settings from "settings"
+import "./Attachment.css"
+import AttachmentCard from "./AttachmentCard"
+import UploadedAttachments from "./UploadedAttachments"
+
+const GQL_CREATE_ATTACHMENT = gql`
+  mutation ($attachment: AttachmentInput!) {
+    createAttachment(attachment: $attachment)
+  }
+`
+
+const GQL_UPDATE_ATTACHMENT = gql`
+  mutation ($attachment: AttachmentInput!) {
+    updateAttachment(attachment: $attachment)
+  }
+`
+
+const UploadAttachment = ({
+  edit,
+  type,
+  attachmentFunc,
+  setIsAttachment,
+  setLoadingError,
+  relatedObjectUuid
+}) => {
+  const base64Marker = ";base64,"
+  const [error, setError] = useState(null)
+  const [remove, setRemove] = useState(false)
+  const [uploadedList, setUploadedList] = useState([])
+
+  // Trigger attachment submit from report save
+  useEffect(() => {
+    attachmentFunc.current = onSubmit
+  })
+
+  const handleUploadFile = file => {
+    setIsAttachment(true)
+    setUploadedList(current => [...current, file])
+    toast.success("Your document has been uploaded")
+  }
+
+  const handleFileEvent = async e => {
+    const file = e.target.files[0]
+    const selectedAttachment = new Attachment()
+    const base64 = await convertBase64(file)
+
+    // Set initial values for new attachment
+    selectedAttachment.content = base64
+    selectedAttachment.fileName = file.name
+    selectedAttachment.mimeType = base64.split(":").pop().split(";")[0]
+    selectedAttachment.attachmentRelatedObjects[0].relatedObjectType = type
+    selectedAttachment.attachmentRelatedObjects[0].relatedObjectUuid = null
+    selectedAttachment.classification =
+      Settings.fields.attachment.classification.choices.UNDEFINED.value
+
+    save(selectedAttachment).then(response => {
+      selectedAttachment.uuid = response.createAttachment
+      selectedAttachment.contentLength = file.size
+      handleUploadFile(selectedAttachment)
+    })
```

Comment:
Maybe add e.g. ` .catch(() => toast.error("Failed to upload your attachment"))` here so at least errors are shown.